include/core/circuit.hpp

```diff
@@ -0,0 +1,1130 @@
+#include <optional>
+#include <utility>
+#include <variant>
+
+namespace syrec {
+    /**
+   * @brief Type for determine whether line is constant or not
+   *
+   * The following table summarizes the use of constant values
+   * in circuit representations.
+   *
+   * <table border="0">
+   *   <tr>
+   *     <td class="indexkey">Description</td>
+   *     <td class="indexkey">Char representation</td>
+   *     <td class="indexkey">Typed value</td>
+   *   </tr>
+   *   <tr>
+   *     <td class="indexvalue">No constant input line</td>
+   *     <td align="center" class="indexvalue">'-'</td>
+   *     <td class="indexvalue">@code constant() @endcode</td>
+   *   </tr>
+   *   <tr>
+   *     <td class="indexvalue">Constant input line with value 0</td>
+   *     <td align="center" class="indexvalue">'0'</td>
+   *     <td class="indexvalue">@code constant( 0 ) @endcode</td>
+   *   </tr>
+   *   <tr>
+   *     <td class="indexvalue">Constant input line with value 1</td>
+   *     <td align="center" class="indexvalue">'1'</td>
+   *     <td class="indexvalue">@code constant( 1 ) @endcode</td>
+   *   </tr>
+   * </table>
+   *
+   * @section Example
+   * This example demonstrates how to access the constant values.
+   * @code
+   * constant c = // some constant
+   *
+   * if ( c ) // checks whether c is set or not
+   * {
+   *   if ( *c ) // c is checked, checks the value of c
+   *   {
+   *     std::cout << "constant value 1" << std::endl;
+   *   }
+   *   else
+   *   {
+   *     std::cout << "constant value 0" << std::endl;
+   *   }
+   * }
+   * else
+   * {
+   *   std::cout << "no constant value" << std::endl;
+   * }
+   * @endcode
+   *
+   * @author RevKit
+   * @since  1.0
+   */
+    typedef std::optional<bool> constant;
+
+    /**
+   * @brief Represents a circuit
+   *
+   * A circuit is represented by a list of gates (type gate::vector)
+   * and meta information like name, inputs, outputs, constants,
+   * and garbage.
+   *
+   * Via STL like iterators the gates can be accessed and also used
+   * in STL and STL like algorithms based on the iterator concept.
+   *
+   * @section example_circuit_class_1 Example: Creating a circuit with 5 lines
+   * @code
+   * #include <core/circuit.hpp>
+   *
+   * ...
+   *
+   * circuit circ( 5 );
+   * @endcode
+   *
+   * @section example_circuit_class_2 Example: Iterate through all gates in a circuit \p circ with iterators
+   * @code
+   * for ( circuit::const_iterator itGate = circ.begin(); itGate != circ.end(); ++itGate )
+   * {
+   *   gate& g = *itGate;
+   * }
+   * @endcode
+   *
+   * @section example_circuit_class_3 Example: Iterator through all gates in a circuit with foreach_
+   * @code
+   * #include <boost/foreach.hpp>
+   * #define foreach_ BOOST_FOREACH
+   *
+   * ...
+   *
+   * foreach_ ( gate& g, circ )
+   * {
+   *   // g can be modified
+   * } 
+   *
+   * foreach_ ( const gate& g, circ )
+   * {
+   *   // g cannot be modified
+   * } 
+   * @endcode
```

Comment:
Should I also remove the doc-strings containing revkit as authors?

Reply:
Depends on whether the doc-strings are still relevant. If they contain references to boost when boost is no longer used there, then these should be fixed.

In my opinion, the `@author RevKit` and `@since ...` parts of the comments can be removed entirely since RevKit is no longer being actively maintained and we are currently building our own tool.